src/pages/reportSighting/ReportForm.jsx

```diff
@@ -276,21 +277,63 @@ export default function ReportForm({
         >
           <Button
             onClick={async () => {
-              // check that required fields are complete
+              // check that required fields are complete.
+              // specifiedTime field is required, but the logic and message
+              // are different from the other fields
               const nextIncompleteFields =
                 defaultSightingSchemas.filter(
                   field =>
                     field.required &&
                     field.defaultValue ===
-                      sightingFormValues[field.name],
+                      sightingFormValues[field.name] &&
+                    field.name !== 'specifiedTime',
                 );
               setIncompleteFields(nextIncompleteFields);
 
+              // check that specifiedTime fields are complete
+              let nextFormErrorId = null;
+              const formTimeSpecificity = get(sightingFormValues, [
+                'specifiedTime',
+                'timeSpecificity',
+              ]);
+              const formTime = get(sightingFormValues, [
+                'specifiedTime',
+                'time',
+              ]);
+
+              const specifiedTimeField =
+                defaultSightingSchemas.find(
+                  field => field.name === 'specifiedTime',
+                ) || {};
+
+              const defaultTimeSpecificity = get(specifiedTimeField, [
+                'defaultValue',
+                'timeSpecificity',
+              ]);
+
+              const defaultTime = get(specifiedTimeField, [
+                'defaultValue',
+                'time',
+              ]);
+
+              if (
+                !formTimeSpecificity ||
+                !formTime ||
+                formTimeSpecificity === defaultTimeSpecificity ||
```

Comment:
It's super easy for the logic here to get really confusing. Can you add parentheses explicitly to make this a little more human readable?

E.g. ,
```
...
  !formTime ||
                (formTimeSpecificity === defaultTimeSpecificity) ||
...
```

Reply:
Prettier removes them when I add them.